src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -614,34 +614,74 @@ def check_for_clashes(self, other_table, symbols_to_skip=()):
             when importing symbols from `other_table` into this table.
 
         '''
+        # pylint: disable-next=import-outside-toplevel
+        from psyclone.psyir.nodes import IntrinsicCall
+
         if not isinstance(symbols_to_skip, Iterable):
             raise TypeError(
                 f"check_for_clashes: 'symbols_to_skip' must be an instance of "
                 f"Iterable but got '{type(symbols_to_skip).__name__}'")
 
+        # Check whether there are any wildcard imports common to both tables.
+        self_imports = self.wildcard_imports()
+        other_imports = other_table.wildcard_imports()
+        shared_wildcard_imports = self_imports & other_imports
+        # Any wildcard imports that appear in one table but not in the other.
+        unique_wildcard_imports = self_imports ^ other_imports
+
         for other_sym in other_table.symbols:
-            if other_sym.name not in self or other_sym in symbols_to_skip:
+            if other_sym.name not in self or other_sym.name in symbols_to_skip:
```

Comment:
This means we can not provide symbols_to_skip that are not shadowed in a inner scope. Don't use .name here.

Reply:
Reverted.